scripts/build-images.sh

```
@@ -0,0 +1,307 @@
+
+    print_status "Waiting for ${#pids[@]} build processes to complete..."
+
+    # Wait for all processes
+    for pid in "${pids[@]}"; do
+        wait "$pid"
+    done
+
+    # Check results
+    for app in "${SPLIT_APPS[@]}"; do
+        if [[ -f "/tmp/build_result_$app" ]]; then
+            local result=$(cat "/tmp/build_result_$app")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)